.github/workflows/test.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install poetry
+      run: |
+        pip install poetry
+    - name: Install dependencies
+      run: |
+        poetry install --with ci 
+
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        poetry run flake8 . --max-line-length=100 --extend-ignore=E203 --per-file-ignores=__init__.py:F401
```

Comment:
instead of running `flake` directly, you could consider reusing your pre-commit hooks so you don't need to keep flags in sync in another place .

 I have in the past used [this action](https://github.com/pre-commit/action). I see is deprecated (but still works). If you don't want to use a deprecated action, you could just use do what that action does, which is call `pre-commit` directly. See [this line](https://github.com/pre-commit/action/blob/4835dd97624f38176d7f9f87fcce089836248ee9/action.yml#L19)